etcdserver/apply.go

```diff
@@ -72,9 +73,17 @@ func (s *EtcdServer) applyV3Request(r *pb.InternalRaftRequest) *applyResult {
 	ar := &applyResult{}
 	switch {
 	case r.Range != nil:
-		ar.resp, ar.err = s.applyV3.Range(noTxn, r.Range)
+		if s.AuthStore().IsRangePermitted(r.UserID, string(r.Range.Key)) {
```

Comment:
range has a start key and end key. we need to check the entire range instead of just one key.